subliminal/providers/legendastv.py

```diff
@@ -359,8 +375,16 @@ def query(self, language, title, season=None, episode=None, year=None):
                     if not a.pack and 'episode' in guess and guess['episode'] != episode:
                         continue
 
+                # calculates a good expiration time
+                if a.timestamp:
+                    delta = max(0, (datetime.datetime.now() - a.timestamp).total_seconds())
+                    # recently uploaded archives have short expiration time
+                    expiration_time = min(RELEASES_EXPIRATION_TIME, delta / 8)
+                else:
+                    expiration_time = RELEASES_EXPIRATION_TIME
+
                 # attempt to get the releases from the cache
-                releases = region.get(releases_key.format(archive_id=a.id))
+                releases = region.get(releases_key.format(archive_id=a.id), expiration_time=expiration_time)
```

Comment:
This is a common misunderstanding. Cache is ever-growing. Nothing is ever deleted from cache. Deleting expired keys isn't part of dogpile.cache's API and never will be. This is why I don't to store binary or whole HTTP requests in cache. With the amount of requests done by subliminal that would mean a really fast paced growing cache.

Only some backends do cache clean up of expired keys based on the backend expiration time which is different from dogpile.cache's.
Documentation says backend's expiration time should be greater than dogpile.cache's but I think a lesser value would trigger a cache refresh, if that's what you're looking for.